rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,242 @@
+
+The idea is that by auto-merging our PRs, we wouldn't have to spend so much time dealing with them. However, there are a few reasons why this isn't possible at the moment:
+
+- We require approvals for all our PRs. Merging in the PR isn't what takes up the time, so having them merge automatically would only shave off a small amount of time.
+- The auto-merge functionality has to be enabled across the entire Dependabot account. We use a shared account across GDS. For this reason, and others, we [weren't able to turn it on][auto-merge-gds-way].
+
+[auto-merge]: https://dependabot.com/blog/automatic-pull-request-merging/
+[auto-merge-discussion]: https://docs.google.com/document/d/18P86EFSZdG8Uv5scbGzVoO8KhA3hkKp80UlzNoIctNE/edit
+[auto-merge-gds-way]: https://github.com/alphagov/gds-way/pull/428#discussion_r426502446
+
+#### Limit updates to lockfile
+
+Another option might be to limit Dependabot to only raise PRs against the `Gemfile.lock` file. The lockfile is used to pin a specific version, whereas the `Gemfile` can contain a wider version range (for example `~> 1.0` means `1.0 all the way up to but not including 2.0`).
+
+Unfortunately applying this would mean [we lose the security updates that do require an update to the `Gemfile`][dependabot-security-lockfile].
+
+[dependabot-security-lockfile]: https://github.com/dependabot/feedback/issues/937
+
+#### Major/minor/patch
+
+This solution proposed above ignores all updates for libraries not in one of the three categories, regardless of whether the update is a major, minor or patch update. An alternative might be to include PRs for major updates for other libraries.
+
+Looking through the Dependabot config docs, [it doesn't seem to be possible to enable major updates][dependabot-allowed-updates]. Regardless, major updates tend to be the ones that take the longest time to merge, so enabling them wouldn't reduce much of the workload.
+
+[dependabot-allowed-updates]: https://dependabot.com/docs/config-file/#allowed_updates
+
+#### Opt-out rather than opt-in
+
+Rather than having a list of libraries we allow updating, we could instead [maintain a list of libraries we ignore updates for][dependabot-ignored-updates].
+
+This would allow us to remove the hardest PRs from updates, but it would require us to be more on top of maintaining the list. I'm also not sure how it would work with enabling security updates regardless.
+
+[dependabot-ignored-updates]: https://dependabot.com/docs/config-file/#ignored_updates
+
+## Consequences and measuring success
+
+Using [the statistics from above](#Statistics), we can make some estimates on what affect this would have had if it had been enabled from the beginning:
+
+- 7378 PRs wouldn't have been raised (10216 rather than the 17526 we've had so far).
```

Comment:
These are really helpful to know! I wonder if we should mention this earlier in the RFC, as it's also an argument for the proposal, as much as a consequence.

Reply:
Good point, I've added that in d28ed61.